src/forms/labels/fieldset-legend.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import { convertNameToLabel } from '../helpers'
+import { useToggle } from '../../utils'
+
+/**
+ *
+ * A dynamic legend associated with a group of inputs within a fieldset.
+ *
+ * This component is used within {@link LabeledFieldset}, and therefore is incorporated into most fieldset grouped input components ({@link RadioGroup}, {@link CheckboxGroup}) by default.
+ *
+ * The text of the legend is set using the following rules:
+ * - If the `label` prop is set to `false`, the legend is hidden completely
```

Comment:
I kept debating whether to change "label" fully to "legend", but I was thinking for backwards compatibility (potentially in code, but also in terms of consistency mentally), sticking with `label` as the prop might be better. It's also still a "label", just a different kind of one. BUT, that's just my two cents.